internal/k8s/reconciler/utils.go

```diff
@@ -204,18 +203,6 @@ func toNamespaceSet(name string, labels map[string]string) klabels.Labels {
 	return klabels.Set(ret)
 }
 
-func sortParents(parents []gw.RouteParentStatus) []gw.RouteParentStatus {
```

Comment:
Since pretty much all of these methods deal with setting status values on the route status, I moved them into a `RouteStatuses` struct that just wraps the underlying map